observability/observability.go

```diff
@@ -0,0 +1,78 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package observability implements the tracing, metrics, and logging data
+// collection, and provides controlling knobs via a config file.
+//
+// Experimental
+//
+// Notice: This package is EXPERIMENTAL and may be changed or removed in a
+// later release.
+package observability
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc/grpclog"
+)
+
+var logger = grpclog.Component("observability")
+
+func init() {
+	prepareLogging()
+}
+
+// Start is the opt-in API for gRPC Observability plugin. This function should
+// be invoked in the main function, and before creating any gRPC clients or
+// servers, otherwise, they might not be instrumented. At high-level, this
+// module does the following:
+//
+//   - it loads observability config from environment;
+//   - it registers default exporters if not disabled by the config;
+//   - it sets up binary logging sink against the logging exporter.
+//
+// Note: this method should only be invoked once.
+// Note: currently, the binarylog module only supports one sink, so using the
+// "observability" module will conflict with existing binarylog usage.
+// Note: handle the error
+func Start(ctx context.Context) error {
+	config, err := parseObservabilityConfig()
+	if err != nil {
+		return err
+	}
+	if config == nil {
+		return fmt.Errorf("no ObservabilityConfig found, it can be set via env %s", envObservabilityConfig)
+	}
+
+	// Set the project ID if it isn't configured manually.
+	maybeUpdateProjectIDInObservabilityConfig(ctx, config)
```

Comment:
We later fail if there's no project ID set.  I think it would be better to fail here, especially when we think about adding other features to observability beyond logging.  Unless project ID isn't required for all observability features?